test/utils/waitUtils.ts

```diff
@@ -0,0 +1,31 @@
+export const waitAndRetry = async <T>(
+  predicateFn: () => T,
+  sleepTime = 100,
+  maxRetryCount = 0,
+): Promise<T> => {
+  return new Promise((resolve, reject) => {
+    let retryCount = 0
+    function performCheck() {
+      if (maxRetryCount !== 0 && retryCount > maxRetryCount) {
+        resolve(predicateFn())
+      }
+      Promise.resolve()
+        .then(() => {
+          return predicateFn()
+        })
+        .then((result) => {
+          if (result) {
+            resolve(result)
+          } else {
+            retryCount++
+            setTimeout(performCheck, sleepTime)
+          }
+        })
+        .catch((err) => {
+          reject(err)
```

Comment:
Shouldn't we retry on errors too?

Reply:
I'm not sure, since this is for using in tests, if we fail with an error somewhere, probably it's fine to fail entire test